app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,21 @@ def show
     @post = @user.posts.find(params[:id])
     @comments = @post.comments
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(params.require(:post).permit(:title, :text))
+    @post.author_id = current_user.id
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      redirect_to user_posts_path(current_user, @post.id), notice: 'Post created successfully.'
+    else
+      flash.now[:error] = 'Error: Post could not be saved'
+      render :new
+    end
+  end
```

Comment:
- [ ] [OPTIONAL] As a suggestion I would like to recommend you add `the params method` , you can read more in this [lesson](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides )  👍 
```ruby
   private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```